### ISSUE TITLE
CreateSeuratObject not picking up designated names.field

### ISSUE THREAD

**Issue by HyacintViolet** [satijalab/seurat#1297]
Hi Seurat team,

I have an expression matrix where the cells are named as BARCODE_BATCH.
I'm trying to input my matrix to CreateSeuratObject, and extract the field BATCH from each cell name.

`> cells<-CreateSeuratObject(countnew, project = "MyProject", min.cells = 10, min.features = 200, names.field = 2, names.delim = '_')`

However in the output, the meta.data$orig.ident field was replaced by project name "MyProject" instead of BATCH.

If I change the names.field argument:
`> cells<-CreateSeuratObject(countnew, project = "MyProject", min.cells = 10, min.features = 200, names.field = 1, names.delim = '_')`
The BARCODES gets extracted correctly, but this isn't what I wanted. I'd like to extract the second field (BATCH).

I also tried changing separators in my matrix to '.' or '-', and change names.delim correspondingly, but same as above, I can extract the BARCODES field, but not BATCH.

I've run the same pipeline on the same data a few month ago and everything worked fine. But then I think I did some updates with R and now it's not working as I expected.

I'm using R 3.5.3 on Ubuntu 18.04 LTS. What might have gone wrong?

Thanks in advance
Best,

Ray

**Comment by timoast** [comment]
Can you provide the output of `head(colnames(countnew))` and `sessionInfo()`?

**Comment by HyacintViolet** [comment]
Hi timoast,

`> cells<-CreateSeuratObject(countnew, project = "Luo", min.cells = 10, min.features = 200, names.field = 2, names.delim = '_')`
`Warning: Feature names cannot have underscores ('_'), replacing with dashes ('-')`

`> head(colnames(countnew))
[1] "AACTCACC_T01" "AAGAGATC_T01" "AAGGACAC_T01" "AATCCGTC_T01" "ACACGACC_T01" "ACAGATTC_T01"`

`R version 3.5.3 (2019-03-11)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 18.04.2 LTS

Matrix products: default
BLAS: /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.7.1
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.7.1

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=zh_CN.UTF-8        LC_COLLATE=en_US.UTF-8     LC_MONETARY=zh_CN.UTF-8   
 [6] LC_MESSAGES=en_US.UTF-8    LC_PAPER=zh_CN.UTF-8       LC_NAME=C                  LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=zh_CN.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] Seurat_3.0.0.9000           dplyr_0.8.0.1               scran_1.10.2                SingleCellExperiment_1.4.1  SummarizedExperiment_1.12.0
 [6] DelayedArray_0.8.0          matrixStats_0.54.0          Biobase_2.42.0              GenomicRanges_1.34.0        GenomeInfoDb_1.18.2        
[11] IRanges_2.16.0              S4Vectors_0.20.1            BiocGenerics_0.28.0         BiocParallel_1.16.6         reshape2_1.4.3             

loaded via a namespace (and not attached):
  [1] Rtsne_0.15               ggbeeswarm_0.6.0         colorspace_1.4-1         ggridges_0.5.1           dynamicTreeCut_1.63-1    XVector_0.22.0          
  [7] BiocNeighbors_1.0.0      rstudioapi_0.10          listenv_0.7.0            npsurv_0.4-0             ggrepel_0.8.0            codetools_0.2-16        
 [13] splines_3.5.3            R.methodsS3_1.7.1        lsei_1.2-0               scater_1.10.1            jsonlite_1.6             ica_1.0-2               
 [19] cluster_2.0.7-1          png_0.1-7                R.oo_1.22.0              HDF5Array_1.10.1         compiler_3.5.3           httr_1.4.0              
 [25] assertthat_0.2.1         Matrix_1.2-15            lazyeval_0.2.2           limma_3.38.3             htmltools_0.3.6          tools_3.5.3             
 [31] rsvd_1.0.0               igraph_1.2.4             gtable_0.3.0             glue_1.3.1               GenomeInfoDbData_1.2.0   RANN_2.6.1              
 [37] Rcpp_1.0.1               gdata_2.18.0             ape_5.3                  nlme_3.1-137             DelayedMatrixStats_1.4.0 gbRd_0.4-11             
 [43] lmtest_0.9-36            stringr_1.4.0            globals_0.12.4           irlba_2.3.3              gtools_3.8.1             statmod_1.4.30          
 [49] future_1.12.0            edgeR_3.24.3             zlibbioc_1.28.0          MASS_7.3-51.1            zoo_1.8-5                scales_1.0.0            
 [55] rhdf5_2.26.2             RColorBrewer_1.1-2       reticulate_1.11.1        pbapply_1.4-0            gridExtra_2.3            ggplot2_3.1.0           
 [61] stringi_1.4.3            caTools_1.17.1.2         bibtex_0.4.2             Rdpack_0.10-1            SDMTools_1.1-221         rlang_0.3.2             
 [67] pkgconfig_2.0.2          bitops_1.0-6             lattice_0.20-38          ROCR_1.0-7               purrr_0.3.2              Rhdf5lib_1.4.2          
 [73] htmlwidgets_1.3          cowplot_0.9.4            tidyselect_0.2.5         plyr_1.8.4               magrittr_1.5             R6_2.4.0                
 [79] gplots_3.0.1.1           pillar_1.3.1             fitdistrplus_1.0-14      survival_2.43-3          RCurl_1.95-4.12          tsne_0.1-3              
 [85] tibble_2.1.1             future.apply_1.2.0       crayon_1.3.4             KernSmooth_2.23-15       plotly_4.8.0             viridis_0.5.1           
 [91] locfit_1.5-9.1           grid_3.5.3               data.table_1.12.0        metap_1.1                digest_0.6.18            tidyr_0.8.3             
 [97] R.utils_2.8.0            munsell_0.5.0            beeswarm_0.2.3           viridisLite_0.3.0        vipor_0.4.5             `

**Comment by HyacintViolet** [comment]
Hi timoast

```
> head(colnames(countnew))
[1] "AACTCACC_T01" "AAGAGATC_T01" "AAGGACAC_T01" "AATCCGTC_T01" "ACACGACC_T01" "ACAGATTC_T01"
```
```
R version 3.5.3 (2019-03-11)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 18.04.2 LTS

Matrix products: default
BLAS: /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.7.1
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.7.1

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=zh_CN.UTF-8        LC_COLLATE=en_US.UTF-8     LC_MONETARY=zh_CN.UTF-8   
 [6] LC_MESSAGES=en_US.UTF-8    LC_PAPER=zh_CN.UTF-8       LC_NAME=C                  LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=zh_CN.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] Seurat_3.0.0.9000           dplyr_0.8.0.1               scran_1.10.2                SingleCellExperiment_1.4.1  SummarizedExperiment_1.12.0
 [6] DelayedArray_0.8.0          matrixStats_0.54.0          Biobase_2.42.0              GenomicRanges_1.34.0        GenomeInfoDb_1.18.2        
[11] IRanges_2.16.0              S4Vectors_0.20.1            BiocGenerics_0.28.0         BiocParallel_1.16.6         reshape2_1.4.3             

loaded via a namespace (and not attached):
  [1] Rtsne_0.15               ggbeeswarm_0.6.0         colorspace_1.4-1         ggridges_0.5.1           dynamicTreeCut_1.63-1    XVector_0.22.0          
  [7] BiocNeighbors_1.0.0      rstudioapi_0.10          listenv_0.7.0            npsurv_0.4-0             ggrepel_0.8.0            codetools_0.2-16        
 [13] splines_3.5.3            R.methodsS3_1.7.1        lsei_1.2-0               scater_1.10.1            jsonlite_1.6             ica_1.0-2               
 [19] cluster_2.0.7-1          png_0.1-7                R.oo_1.22.0              HDF5Array_1.10.1         compiler_3.5.3           httr_1.4.0              
 [25] assertthat_0.2.1         Matrix_1.2-15            lazyeval_0.2.2           limma_3.38.3             htmltools_0.3.6          tools_3.5.3             
 [31] rsvd_1.0.0               igraph_1.2.4             gtable_0.3.0             glue_1.3.1               GenomeInfoDbData_1.2.0   RANN_2.6.1              
 [37] Rcpp_1.0.1               gdata_2.18.0             ape_5.3                  nlme_3.1-137             DelayedMatrixStats_1.4.0 gbRd_0.4-11             
 [43] lmtest_0.9-36            stringr_1.4.0            globals_0.12.4           irlba_2.3.3              gtools_3.8.1             statmod_1.4.30          
 [49] future_1.12.0            edgeR_3.24.3             zlibbioc_1.28.0          MASS_7.3-51.1            zoo_1.8-5                scales_1.0.0            
 [55] rhdf5_2.26.2             RColorBrewer_1.1-2       reticulate_1.11.1        pbapply_1.4-0            gridExtra_2.3            ggplot2_3.1.0           
 [61] stringi_1.4.3            caTools_1.17.1.2         bibtex_0.4.2             Rdpack_0.10-1            SDMTools_1.1-221         rlang_0.3.2             
 [67] pkgconfig_2.0.2          bitops_1.0-6             lattice_0.20-38          ROCR_1.0-7               purrr_0.3.2              Rhdf5lib_1.4.2          
 [73] htmlwidgets_1.3          cowplot_0.9.4            tidyselect_0.2.5         plyr_1.8.4               magrittr_1.5             R6_2.4.0                
 [79] gplots_3.0.1.1           pillar_1.3.1             fitdistrplus_1.0-14      survival_2.43-3          RCurl_1.95-4.12          tsne_0.1-3              
 [85] tibble_2.1.1             future.apply_1.2.0       crayon_1.3.4             KernSmooth_2.23-15       plotly_4.8.0             viridis_0.5.1           
 [91] locfit_1.5-9.1           grid_3.5.3               data.table_1.12.0        metap_1.1                digest_0.6.18            tidyr_0.8.3             
 [97] R.utils_2.8.0            munsell_0.5.0            beeswarm_0.2.3           viridisLite_0.3.0        vipor_0.4.5             
```

**Comment by timoast** [comment]
I was unable to reproduce this, could you send a downsampled version of your `countnew` matrix that reproduces this error?

**Comment by HyacintViolet** [comment]
Hi timoast,

I think I know what my problem was.
Previously I concatenated all my libraries (108 in total) into a matrix. I wasn't sure how to properly down-sample the matrix so I dropped half of them, leaving 54 libraries in the matrix.

Then I called CreateSeuratObject on the down-sampled matrix and everything worked fine. I guess the issue I encountered has to do with my memory size.
I'll try dividing my libraries into smaller pieces and load them individually into Seurat object, which may solve the problem.

Best,
Ray
